xl2times/transforms.py

```diff
@@ -222,6 +222,44 @@ def discard(table):
     return result
 
 
+def revalidate_input_tables(
+    config: datatypes.Config,
+    tables: List[datatypes.EmbeddedXlTable],
+    model: datatypes.TimesModel,
+) -> List[datatypes.EmbeddedXlTable]:
+    """
+    Perform further validation of input tables by checking whether required columns are
+    present / non-empty. Remove tables without required columns or if they are empty.
+    """
+    result = []
+    for table in tables:
+        tag = datatypes.Tag(table.tag.split(":")[0])
+        required_cols = config.required_columns[tag]
+        unique_table_cols = set(table.dataframe.columns)
+        if required_cols:
+            # Drop table if any column in required columns is missing
+            if not required_cols.issubset(unique_table_cols):
+                missing_cols = required_cols - unique_table_cols
```

Comment:
```suggestion
            missing_cols = required_cols - unique_table_cols
            if missing_cols:
```
Avoids recomputing the set difference :)